internal/services/pam/pam_test.go

```diff
@@ -0,0 +1,596 @@
+package pam_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd"
+	"github.com/ubuntu/authd/internal/brokers"
+	"github.com/ubuntu/authd/internal/services/pam"
+	"github.com/ubuntu/authd/internal/testutils"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+var (
+	generatedID string
+	testClient  authd.PAMClient
+)
+
+// Used for TestGetAuthenticationModes and TestSelectAuthenticationMode.
+var requiredEntries = "required:entry_type,other_entry_type"
+var optionalEntries = "optional:entry_type,other_entry_type"
+var optional = "optional"
+var requiredEntry = &authd.UILayout{
+	Type:    "required-entry",
+	Label:   &optional,
+	Button:  &optional,
+	Wait:    &optional,
+	Entry:   &requiredEntries,
+	Content: &optional,
+}
+var optionalEntry = &authd.UILayout{
+	Type:  "optional-entry",
+	Entry: &optionalEntries,
+}
+var emptyType = &authd.UILayout{
+	Type:  "",
+	Entry: &requiredEntries,
+}
+
+func TestAvailableBrokers(t *testing.T) {
+	t.Parallel()
+
+	abResp, err := testClient.AvailableBrokers(context.Background(), &authd.Empty{})
+	require.NoError(t, err, "AvailableBrokers should not return an error, but did")
+
+	got := abResp.GetBrokersInfos()
+	for _, broker := range got {
+		broker.Id = broker.Name + "_ID"
+	}
+	want := testutils.LoadWithUpdateFromGoldenYAML(t, got)
+	require.Equal(t, want, got, "AvailableBrokers returned unexpected brokers")
+}
+
+func TestSetDefaultBrokerForUser(t *testing.T) {
```

Comment:
another request: do you mind trying to keep approximately the same order than the one in the PAM module? The order mostly follows the order of requests (but not completely). At least, from most important to least. I think it will be good to follow the same idea in the tests.